cli/filplus.go

```diff
@@ -162,22 +169,51 @@ var filplusListClientsCmd = &cli.Command{
 	Name:  "list-clients",
 	Usage: "list all verified clients",
 	Action: func(cctx *cli.Context) error {
+		if cctx.NArg() != 0 {
+			return IncorrectNumArgs(cctx)
+		}
+
 		api, closer, err := GetFullNodeAPI(cctx)
 		if err != nil {
 			return err
 		}
 		defer closer()
 		ctx := ReqContext(cctx)
 
-		act, err := api.StateGetActor(ctx, verifreg.Address, types.EmptyTSK)
+		apibs := blockstore.NewAPIBlockstore(api)
+		store := adt.WrapStore(ctx, cbor.NewCborStore(apibs))
+
+		nv, err := api.StateNetworkVersion(ctx, types.EmptyTSK)
 		if err != nil {
 			return err
 		}
 
-		apibs := blockstore.NewAPIBlockstore(api)
-		store := adt.WrapStore(ctx, cbor.NewCborStore(apibs))
+		av, err := actorstypes.VersionForNetwork(nv)
+		if err != nil {
+			return err
+		}
 
-		st, err := verifreg.Load(store, act)
+		if av <= 8 {
```

Comment:
This `if av <= 8`, do the thing through verifreg, if not do the thing through datacap pattern is gonna become common.

Do you think it'd be feasible to have some abstractor.......somewhere...that does this for you? You give it the `ForEach` func, and it looks up the appropriate state and applies it?

Can be chucked into a P3 issue if you don't feel inspired.

Reply:
Definitely not feeling super inspired to do this.

Reply:
Yup, do open an issue for it, though

Reply:
https://github.com/filecoin-project/lotus/issues/9438